src/TDebug.cpp

```diff
@@ -122,9 +156,117 @@ TDebug& TDebug::operator<<(const QList<QString>& list)
 TDebug& TDebug::operator<<(const QList<int>& list)
 {
     for (QList<int>::const_iterator it = list.constBegin(); it != list.constEnd(); ++it) {
-        msg += QString::number(*it);
-        msg += ", ";
+        msg += QStringLiteral("%1, ").arg(QString::number(*it));
     }
     msg += "), ";
     return *this;
 }
+
+void TDebug::changeHostName(const Host* pHost, const QString& newName)
+{
+    if (pHost) {
+        QPair<QString, QChar>& pair = TDebug::smIdentifierMap[pHost];
+        pair.first = newName;
+    }
+}
+
+/* static */ void TDebug::addHost(Host* pHost)
+{
+    if (!pHost) {
+        return;
+    }
+    static const QString validIdentifiers{"123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"};
+    static const int maxIdCount = validIdentifiers.size();
+    int tryCount = -1;
+    do {
+        if (!TDebug::smUsedIdentifiers.contains(validIdentifiers.at(++tryCount))) {
+            break;
+        }
+    } while (tryCount < maxIdCount);
+
+    QString hostName = pHost->getName();
+    if (TDebug::smUsedIdentifiers.contains(validIdentifiers.at(tryCount))) {
+        // Run out of identifers - fall back to '?'
+        QPair<QString, QChar> newIdentifier = qMakePair(hostName, QLatin1Char('?'));
+        TDebug::smIdentifierMap.insert(pHost, newIdentifier);
+    } else {
+        QPair<QString, QChar> newIdentifier = qMakePair(hostName, validIdentifiers.at(tryCount));
+        TDebug::smUsedIdentifiers.insert(validIdentifiers.at(tryCount));
+        TDebug::smIdentifierMap.insert(pHost, newIdentifier);
+    }
+    TDebug localMessage(Qt::blue, Qt::white);
+    localMessage << QStringLiteral("Profile: '%1' started.\n").arg(hostName) >> nullptr;
```

Comment:
```suggestion
    localMessage << QStringLiteral("Profile '%1' started.\n").arg(hostName) >> nullptr;
```
Remove punctuation in sentence middle

Reply:
:+1: :-1: This is a bit subjective IMHO - but I *could* live without the ':' I suppose...

Reply:
Please do, or otherwise show sentences like "street: sesame built." It's just wrong grammar, no subjectivity required.

Reply:
*I beg to differ **:¹*** https://christopheraltman.wordpress.com/tag/colon-basics-punctuation-toolbox/ is one person's attempt to introduce it as an *introducer*...

However I am doing it in the next commit,

***¹** - Due to some oddity in the GH markdown I have to put a space before that `:` otherwise it won't enable me to embolden it and not the preceding characters!*

Reply:
I like to discuss grammar, and the examples you show as introducing are always preceded by what could otherwise stand as a sentence of its own. I would agree if your sentence read "street built: 'sesame'" but not as it was. Thanks for removing that then.

Reply:
For comparison: https://writingcenter.unc.edu/tips-and-tools/semi-colons-colons-and-dashes/#:~:text=How%20to%20check%20for%20mistakes

> How to check for mistakes
Ask yourself a question: does the material preceding the colon stand on its own? One way to tell if the colon has been properly used is to look only at the words that come in front of the colon. Do they make a complete thought? If not, you may be using the colon improperly. Check above to see if you have made one of the most common mistakes.

Where number 1 was "Using a colon between a verb and its object"

Source: University of North Carolina